### PR TITLE
feat: implement-crud-backend

### PR DESCRIPTION
**Pull request by Khadgarrm**
crud back

### CODE REVIEW

File: server/src/controllers/todo.controller.ts [hunk 1]
```diff
@@ -1,13 +1,56 @@
...
-    constructor(private todoService: TodoService) {}
+    constructor(private todoService: TodoService) { }
+
     async getAllTodo(_: Request, res: Response) {
-        // TODO: Write your implementation here
-        const todos = await this.todoService.findAll();
-        res.send(todos);
+
+        const err = validationResult(_)
+        if (!err.isEmpty()) {
```
Comment (CodeGeneration-2020) [inline server/src/controllers/todo.controller.ts:11]: Do not dublicate code with error handler move that logic into ValidatorClass

File: server/src/controllers/todo.controller.ts [hunk 2]
```diff
@@ -1,13 +1,56 @@
...
+
+            const todos = await this.todoService.findAll();
+            res.status(200).json(todos);
+    }
+    
+    async createTodo(_: Request, res: Response) {
+
+        const err = validationResult(_)
+        if (!err.isEmpty()) {
+            return res.status(400).json({
```
Comment (CodeGeneration-2020) [inline server/src/controllers/todo.controller.ts:26]: Do not dublicate code with error handler move that logic into ValidatorClass

File: server/src/controllers/todo.controller.ts [hunk 3]
```diff
@@ -1,13 +1,56 @@
...
+        }
+
+        const todo = await this.todoService.createOne(_, res)
+        res.json(todo)
     }
+
+    async updateTodo(_: Request, res: Response) {
+        const err = validationResult(_)
+        if (!err.isEmpty()) {
+            return res.status(400).json({
```
Comment (CodeGeneration-2020) [inline server/src/controllers/todo.controller.ts:39]: Do not dublicate code with error handler move that logic into ValidatorClass

File: server/src/controllers/todo.controller.ts [hunk 4]
```diff
@@ -1,13 +1,56 @@
...
+                message: "Something went wrong."
+            })
+        }
+
+        const todo = await this.todoService.updateOne(_, res)
+        res.json(todo)
+    }
+
+    async removeTodo(_: Request, res: Response) {
+        const todo = await this.todoService.removeOne(_.params.id)
```
Comment (CodeGeneration-2020) [inline server/src/controllers/todo.controller.ts:50]: check fir existance here

File: server/src/routes/api/todos.route.ts [hunk 5]
```diff
@@ -1,9 +1,28 @@
...
 import todoController from "../../controllers/todo.controller";
 
-const todosRouter: Router = Router();
+const todoRouter: Router = Router();
 
-todosRouter.get("", todoController.getAllTodo.bind(todoController));
+todoRouter.get("", todoController.getAllTodo.bind(todoController));
 
-export default todosRouter;
+todoRouter.post("", [
```
Comment (CodeGeneration-2020) [inline server/src/routes/api/todos.route.ts:10]: move that validation check into validatr class

File: server/src/routes/api/todos.route.ts [hunk 6]
```diff
@@ -1,9 +1,28 @@
...
+    check("title", " Fill in the required field").not().isEmpty(),
+    check("description", " Fill in the required field").not().isEmpty(),
+    check("date", " Fill in the required field").not().isEmpty().isNumeric(),
+    check("isPublic").toBoolean(),
+    check("iscompleted").toBoolean(),
+], todoController.createTodo.bind(todoController))
+
+todoRouter.put("/:id", [
+    check("title", " Fill in the required field").not().isEmpty(),
+    check("description", " Fill in the required field").not().isEmpty(),
```
Comment (CodeGeneration-2020) [inline server/src/routes/api/todos.route.ts:20]: Code dublicate

File: server/src/server.ts [hunk 7]
```diff
@@ -50,4 +50,14 @@ const server = app.listen(port, () =>
   console.log(`Server started on port ${port}`)
 );
 
+const todo = [
+  {
+    title: 'dfsfs',
+    description: 'dfsf',
```
Comment (CodeGeneration-2020) [inline server/src/server.ts:56]: ??

File: server/src/services/todo.service.ts [hunk 8]
```diff
@@ -1,5 +1,46 @@
...
+
+        const todo = await Todo.findOneAndUpdate({ _id: req.params.id }, {
+            title: title,
+            description: description,
+            date: date,
+            isPublic: isPublic,
+            iscompleted: isCompleted
+        })
+
+        await todo!.save()
```
Comment (CodeGeneration-2020) [inline server/src/services/todo.service.ts:38]: yo do not need to check here for existance

Comment (Khadgarrm) [reply server/src/services/todo.service.ts:38]: eslint complain that object is possibly equel "null"

**Pull request merged**